worlds/pokemon_emerald/__init__.py

```diff
@@ -0,0 +1,811 @@
+                picked_evolution = starter.species_id
+                if len(potential_evolutions) > 0:
+                    picked_evolution = self.random.choice(potential_evolutions)
+
+                for trainer_name, starter_position, is_evolved in rival_teams[i]:
+                    trainer_data = self.modified_data.trainers[emerald_data.constants[trainer_name]]
+                    trainer_data.party.pokemon[starter_position].species_id = picked_evolution if is_evolved else starter.species_id
+
+        self.modified_data = copy.deepcopy(emerald_data)
+
+        # Randomize species data
+        if self.options.abilities != RandomizeAbilities.option_vanilla:
+            randomize_abilities()
+
+        if self.options.types != RandomizeTypes.option_vanilla:
+            randomize_types()
+
+        if self.options.level_up_moves != LevelUpMoves.option_vanilla:
+            randomize_learnsets()
+
+        randomize_tm_hm_compatibility()  # Options are checked within this function
+
+        min_catch_rate = min(self.options.min_catch_rate.value, 255)
+        for species in self.modified_data.species:
+            if species is not None:
+                species.catch_rate = max(species.catch_rate, min_catch_rate)
+
+        if self.options.tm_moves:
+            randomize_tm_moves()
+
+        # Randomize wild encounters
+        if self.options.wild_pokemon != RandomizeWildPokemon.option_vanilla:
+            randomize_wild_encounters()
+
+        # Randomize static encounters
+        if self.options.static_encounters != RandomizeStaticEncounters.option_vanilla:
+            randomize_static_encounters()
+
+        # Randomize opponents
+        if self.options.trainer_parties != RandomizeTrainerParties.option_vanilla:
+            randomize_opponent_parties()
+
+        # Randomize starters
+        if self.options.starters != RandomizeStarters.option_vanilla:
+            randomize_starters()
+
+        generate_output(self, output_directory)
+
+    def fill_slot_data(self) -> Dict[str, Any]:
+        slot_data: Dict[str, Any] = {}
+
+        sent_options = [
+            "goal",
+            "badges",
+            "hms",
+            "key_items",
+            "bikes",
+            "rods",
+            "overworld_items",
+            "hidden_items",
+            "npc_gifts",
+            "require_itemfinder",
+            "require_flash",
+            "enable_ferry",
+            "elite_four_requirement",
+            "elite_four_count",
+            "norman_requirement",
+            "norman_count",
+            "extra_boulders",
+            "free_fly_location",
+            "fly_without_badge",
+        ]
+
+        for option_name in sent_options:
+            option = getattr(self.options, option_name)
+            slot_data[option_name] = option.value
+
+        slot_data["free_fly_location_id"] = self.free_fly_location_id
+        slot_data["remove_roadblocks"] = list(self.options.remove_roadblocks.value)
```

Comment:
python sets are json serializable so this isn't necessary, and can also be added to the `as_dict` call.

Reply:
Changed in c061c9c (and #2354)